thijsvanloef/palworld-server-docker
# Cron backup

## Context 

* Adds supercronic for scheduled backups
* Will make it easier to implement #187 & #37
* Sets a backup for everyday at 12:00am (Based on TZ set)
* Continuing the work of @GuillaumeBarreau from #101
* Addressing #39 

## Choices

* Set as default to prevent issues like #183, Although I think it should be ran hourly but I will wait until #188 as to not eat up storage.
* Using supercronic instead of cron since cron is not able to use environmental variables and it outputs to terminal
`palworld-server-test  | time="2024-01-28T17:33:00-05:00" level=info msg=starting iteration=0 job.command="bash /usr/local/bin/backup" job.position=0 job.schedule="* * * * *"`

## Test instructions

1. Set `BACKUP_CRON_EXPRESSION` to `* * * * *` to backup every minute to reduce waiting while testing
2. Verify backups are being made every minute
3. Verify supercronic starts as steam: `docker exec -it palworld-server bash -c "top -n 1 | grep supercronic | grep --color steam"`

## Checklist before requesting a review
